src/ansible_navigator/configuration_subsystem/utils.py

```diff
@@ -0,0 +1,174 @@
+"""Utilities related to the configuration subsystem."""
+
+import sys
+
+from types import SimpleNamespace
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+from .definitions import ApplicationConfiguration
+from .definitions import CliParameters
+from .definitions import Constants as C
+from .definitions import SettingsEntry
+
+
+class HumanReadableEntry(NamedTuple):
```

Comment:
Why not use a dataclass? Also, this doesn't seem to belong in an abstract "utils" module, maybe it'd be better off in a separate "data structures" module. In fact, having a wholesale "utils" module is an antipattern — try to come up with a more accurate name for what these utils do and if there's no one common nameable purpose, put things in separate modules with clear names. Here's some information on why this is important: https://tonsky.me/blog/utils/.

Reply:
In this case a mutable structure is not required, and this particular definition is limited to and should remain limited to this file.  It exists only as a way to more sanely populate the dictionaries being returned.

Reply:
Well, it has a "public" name (w/o a leading underscore). And it's still better be a dataclass. Just don't forget to use `@dataclass(frozen=True, slots=True)` for this.

Reply:
Good call, the name should be _HumanReadableEntry.

Can you explain why a dataclass is better here?

Reply:
In general, dataclasses are a better default: https://www.python.org/dev/peps/pep-0557/#why-not-just-use-namedtuple. If this is not a tuple semantically, then we wouldn't want all of the tuple properties.

Reply:
That's a convincing read.  Let's change it.